ansible-sidecar/app/server.py

```
@@ -0,0 +1,181 @@
+
+    run_id = payload.get("runId")
+    if not isinstance(run_id, str) or not re.fullmatch(r"[A-Za-z0-9_-]{1,64}", run_id):
+        raise InvalidRequest("runId must match [A-Za-z0-9_-]{1,64}")
+
+    playbook = payload.get("playbook")
+    if not isinstance(playbook, str) or not _PLAYBOOK_NAME.fullmatch(playbook):
+        raise InvalidRequest("playbook must be a bare .yml filename")
+
+    hosts = payload.get("hosts")
+    if not isinstance(hosts, list) or not hosts or not all(isinstance(h, str) and h for h in hosts):
+        raise InvalidRequest("hosts must be a non-empty array of strings")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)